flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -500,8 +500,9 @@ private void setVertexConfig(Integer vertexID, StreamConfig config,
 
 		config.setStateBackend(streamGraph.getStateBackend());
 		config.setCheckpointingEnabled(checkpointCfg.isCheckpointingEnabled());
-		config.setUnalignedCheckpointsEnabled(checkpointCfg.isUnalignedCheckpointsEnabled());
 		config.setCheckpointMode(getCheckpointingMode(checkpointCfg));
+		config.setUnalignedCheckpointsEnabled(config.getCheckpointMode() == CheckpointingMode.EXACTLY_ONCE &&
```

Comment:
Isn't this an independent bug fix?

Reply:
So far, we have assumed that the user only sets UC when checkpoint mode is exactly once. In this PR, we ignoring checkpoint mode and set it for all.

The question is where to address that:
- We could fail on that combination. Then we would need to disable UC for every at least once test. Not my favorite.
- We change checkpointing mode with a warning. Not my favorite.
- We disable/ignore UC (current way) with a warning (needs to be added).
- We could have a value `UC if exactly once` and make `checkpointCfg.isUnalignedCheckpointsEnabled()` more intelligent. Would also be plausible.

Reply:
I guess I would prefer the most:
> We could fail on that combination. Then we would need to disable UC for every at least once test. Not my favorite.

But this one is probably also fine:
> We disable/ignore UC (current way) with a warning (needs to be added).

But either way, I think this should deserve it's own ticket.

Reply:
Extracted FLINK-18403, which will be backported (while this PR should only go to master).